src/UI/Component/Triggerable.php

```diff
@@ -0,0 +1,24 @@
+<?php
+namespace ILIAS\UI\Component;
+
+/**
+ * Interface Triggerable
+ *
+ * Describes a component offering signals that can be triggered by other components on events.
+ * Example: A modal offers signals to show and close the modal.
+ *
+ * A signal is represented by a unique string identifier and may offer some options which can be passed
+ * by a triggerer component when triggering the signal.
```

Comment:
... or this. Probably forget may comment above about the need of describing the signal, or maybe copy this description.